homeassistant/components/zha/api.py

```diff
@@ -585,6 +634,65 @@ def async_load_api(hass):
         schema=SERVICE_SCHEMAS[SERVICE_ISSUE_ZIGBEE_CLUSTER_COMMAND],
     )
 
+    async def warning_device_squawk(service):
+        """Issue the squawk command for an IAS warning device."""
+        ieee = service.data.get(ATTR_IEEE)
+        mode = service.data.get(ATTR_WARNING_DEVICE_MODE)
+        strobe = service.data.get(ATTR_WARNING_DEVICE_STROBE)
+        level = service.data.get(ATTR_LEVEL)
+
+        zha_device = zha_gateway.get_device(ieee)
+        if zha_device is not None:
+            channel = zha_device.cluster_channels.get(CHANNEL_IAS_WD)
+            if channel:
+                await channel.squawk(mode, strobe, level)
+        _LOGGER.debug(
```

Comment:
@MartinHjelmare if you happen to have a moment... is there a standard not found exception for services like there is for web socket commands? Or, is there a preferred way to deal with not found exceptions in services?

Reply:
We don't have a standard exception for that, I believe. We normally just log error and return. We try to avoid leaking errors out of the integration.

There's `ServiceNotFound` exception that the core raises when a service is called that doesn't exist. But we'd need a different error here.

Reply:
Thanks, I’ll just log and return.

Reply:
@MartinHjelmare but if services raises an exception, then user gets a real time notification about service call failure. Quite useful, if you need  immediate feedback. With logs, users have to check the logs, which is less than ideal UX.

Reply:
I'm talking about 
![image](https://user-images.githubusercontent.com/5826160/65333850-3a834b00-db8f-11e9-976a-8742546ad795.png)

Reply:
Yeah, I think we should have this discussion in general first before we change any integration to that pattern.

Please make an architecture issue.

Reply:
Done: https://github.com/home-assistant/architecture/issues/291